packages/language-server/src/cache.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  ManifestDetails,
+  UI5SemanticModel,
+  YamlDetails,
+} from "@ui5-language-assistant/semantic-model-types";
+
+type AbsolutePathToManifest = string;
+type AbsolutePathToYaml = string;
+class ProjectCache {
+  private _manifestDetails: Map<AbsolutePathToManifest, ManifestDetails>;
+  private _ui5YamlDetails: Map<AbsolutePathToYaml, YamlDetails>;
+  private _ui5Model: Map<string, UI5SemanticModel>;
+  constructor() {
+    this._manifestDetails = new Map();
+    this._ui5YamlDetails = new Map();
+    this._ui5Model = new Map();
+  }
+
+  /**
+   * Get all cached manifest details
+   */
+  getAllManifestDetails(): Map<AbsolutePathToManifest, ManifestDetails> {
```

Comment:
I'm not sure we need all these getters and setters. 
They usually not needed unless we have some internal data transformation/validation inside the class.
You allow to access _manifestDetails directly and at the same time offer to update it via setManifestDetails method. IMHO it is fake security. Much simpler just to have public class member

Reply:
well we are not very much concern about security here, but anyway to to make private class members not to be publicly exposed, redefine it. Thanks anyway.